src/coreclr/tools/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -3306,6 +3307,46 @@ private uint getClassDomainID(CORINFO_CLASS_STRUCT_* cls, ref void* ppIndirectio
             }
         }
 
+#pragma warning disable CA1822 // Mark members as static
+        private bool getReadonlyStaticFieldValue(CORINFO_FIELD_STRUCT_* fieldHandle, byte* buffer, int bufferSize)
+#pragma warning restore CA1822 // Mark members as static
+        {
+            FieldDesc field = HandleToObject(fieldHandle);
+            if (field.IsStatic && !field.IsThreadStatic && field.IsInitOnly && !field.IsIntrinsic && !field.HasGCStaticBase &&
```

Comment:
```suggestion
            if (field.IsStatic && !field.IsThreadStatic && field.IsInitOnly) 
```
This should be left for `PreinitializationManager.IsPreinitialized` to figure out.